core/sql/common/MiscType.h

```diff
@@ -106,25 +118,63 @@ class SQLBoolean : public NAType
 				       CollHeap* h,
 				       UInt32 *flags = NULL) const;
   
+  // ---------------- Methods not inherited from NAType ----------------
+
+  NABoolean canBeSQLUnknown() const { return sqlUnknownFlag_;}
+
+  private:
+  NABoolean sqlUnknownFlag_;
```

Comment:
So, the reason SQLBooleanRelat is a different class than SQLBooleanNative is that SQLBooleanRelat has to model the three-valued logic of SQL predicates (that is, true/false/unknown), while SQLBooleanNative has to model a possibly nullable two-value logic (true/false/null).